### PR TITLE
feat(api): initialize metric tracking

### PR DESCRIPTION
**Pull request by Lee-W**
In this PR, we track "request id", "score", 'success" and "created_at" which can be used to calculate the number of traces (i.e. root runs = user requests) and another view will be created to group requst by success/failure, and the calculate Avg. correctness score, by day.

closes: #196

We do not have Snowflake access, and the Snowflake trial didn't work for me. Use SQLite as a POC. will change it to snowflake after we get the access

### CODE REVIEW

**Review by sunank200 — COMMENTED** [review]
@Lee-W could you update the PR description with all the metrics you are tracking, please?

I would suggest to add it in detail

**Comment by Lee-W** [comment]
image


image

File: api/ask_astro/config.py [hunk 1]
```diff
@@ -61,3 +61,13 @@ class WeaviateConfig:
     index_name = os.environ.get("WEAVIATE_INDEX_NAME")
     text_key = os.environ.get("WEAVIATE_TEXT_KEY")
     attributes = os.environ.get("WEAVIATE_ATTRIBUTES", "").split(",")
+
+
+class MetricsDBConfig:
+    """Containers the config variables for Metrics DB"""
+
+    user = os.environ.get("METRICS_DB_USER")
```
Comment (sunank200) [inline api/ask_astro/config.py:69]: should we name it `METRICS_SNOWFLAKE_DB_USER`? and change others as well?

File: tasks/api.py [hunk 2]
```diff
@@ -75,3 +75,60 @@ def test(ctx: Context) -> None:
...
+    create_request_table_sql = f"""
+        CREATE TABLE IF NOT EXISTS {database}.{schema}.request(
+            uuid TEXT,
+            score INT DEFAULT 0,
+            success BOOLEAN,
+            created_at timestamp default current_timestamp(),
+            PRIMARY KEY (uuid)
+        );
+    """
+    print(create_request_table_sql)
```
Comment (sunank200) [inline tasks/api.py:92]: change this to logging

File: tasks/api.py [hunk 3]
```diff
@@ -75,3 +75,60 @@ def test(ctx: Context) -> None:
...
+                success,
+                score,
+                date_trunc('DAY',created_at) as created_at_day
+            FROM
+                {database}.{schema}.request
+        )
+        GROUP BY
+            created_at_day, success;
+    """
+    print(create_request_summary_view_sql)
```
Comment (sunank200) [inline tasks/api.py:117]: change this to logging

**Review by sunank200 — CHANGES_REQUESTED** [review]
Why are there files like api.pickle, parquet files, csv files added as part of this PR?

File: api/ask_astro/config.py (hunk 1 above)
Comment (Lee-W) [reply api/ask_astro/config.py]: We could. Just rename it. Thanks!

File: tasks/api.py (hunk 2 above)
Comment (Lee-W) [reply tasks/api.py]: Just changed it. Thanks!

File: tasks/api.py (hunk 3 above)
Comment (Lee-W) [reply tasks/api.py]: Just changed it. Thanks!

**Comment by Lee-W** [comment]
> Why are there files like api.pickle, parquet files, csv files added as part of this PR?

They're accidentally added. Removed them.

**Review by pankajastro — COMMENTED** [review]
this looks good to me. But some similar data get stored in Firestore so it feels like redundant data between Snowflake and Firestore. Also from the maintenance point of view if we use only one between  Firestore and Snowflake that would be great.

**Comment by sunank200** [comment]
> this looks good to me. But some similar data get stored in Firestore so it feels like redundant data between Snowflake and Firestore. Also from the maintenance point of view if we use only one between Firestore and Snowflake that would be great.

Thats right. But this is a request from Steven for data team

**Comment by Lee-W** [comment]
Just rewrite it  as airflow DAG as informed

File: airflow/dags/metrcis/load_firestore_to_snowflake.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+from datetime import date, datetime, timedelta
+
+import snowflake.connector
+from google.cloud import firestore
+from google.cloud.firestore_v1.base_query import FieldFilter
+from include.utils.slack import send_failure_notification
+
+from airflow.decorators import dag, task
+
+METRICS_SNOWFLAKE_DB_USER = os.environ.get("METRICS_SNOWFLAKE_DB_USER")
```
Comment (sunank200) [inline airflow/dags/metrcis/load_firestore_to_snowflake.py:17]: Should we not get these from the snowflake connection of airflow instead?

Comment (Lee-W) [reply airflow/dags/metrcis/load_firestore_to_snowflake.py:17]: We could, but I'm not sure whether we want. We'll need to install the provider simply to get the conn.

Comment (sunank200) [reply airflow/dags/metrcis/load_firestore_to_snowflake.py:17]: Could we not use this operator to run the query: https://airflow.apache.org/docs/apache-airflow-providers-snowflake/stable/operators/snowflake.html#snowflakesqlapioperator?

Comment (sunank200) [reply airflow/dags/metrcis/load_firestore_to_snowflake.py:17]: But I am okay with current implementation for now but we could track this in future PRs.

**Review by sunank200 — APPROVED** [review]
LGTM. Let's make sure to add a section in this doc for this [here](https://www.notion.so/astronomerio/Ask-Astro-Monitoring-Observability-d22964e6166f4580837e924339cedf16) before we merge this PR.

**Comment by Lee-W** [comment]
Just update the documentation. Will merge this PR now

**Pull request merged**